lib/tasks/openapi_generate.rake

```diff
@@ -31,11 +31,12 @@ class OpenapiGenerator < Insights::API::Common::OpenApi::Generator
     end
   end
 
-  def schema_overrides
-    authentication = schemas['Authentication']
-    authentication['properties']['authtype']['readOnly'] = true
-    super.merge('Authentication' => authentication)
-  end
+  # TODO: make readOnly authtype only for update
+  # def schema_overrides
+  #   authentication = schemas['Authentication']
+  #   authentication['properties']['authtype']['readOnly'] = true
+  #   super.merge('Authentication' => authentication)
+  # end
```

Comment:
I'm curious your opinion on this @gmcculloug, should we just `git revert 23677837939866329a48048044330ef42cb972f0` (the prior commit) for now until we can implement it in the openapi.json?

Personally I'd rather implement it there rather than having a validation for it.

Reply:
@lindgrenj6 I'm really not sure what is the best approach here.  As we briefly chatted about it could also be handled at the controller layer during the update call as well.

There seem to be 3 independent solutions:
1. Update OpenAPI spec to use a different object for Patch calls that define authtype as read-only.  Requires work on auto-generator
2. Model validation (this PR)
3. Limit parameters during update on the controller.

@lindgrenj6 What is the concern about having the validation?

@mkanoor Could use your thoughts on this.

Reply:
I don't really have a concern with the validation itself, I would just like to solve this in the OpenAPI spec which as you said requires work on the auto-generator. 

Maybe we should just make a git-issue or jira ticket that we should enhance the auto-generator to handle different objects for POST requests, that way we can track it.

Reply:
@lindgrenj6 with https://projects.engineering.redhat.com/browse/TPINVTRY-1004 as our way to solve the issues found using the auto-generator. Do you feel the concerns for now are mitigated?